tests/publiccloud/img_proof.pm

```diff
@@ -27,33 +27,45 @@ our $gce_byos      = 'test_sles_wait_on_registration,test_sles,test_sles_gce';
 our $gce_on_demand = 'test_sles_wait_on_registration,test_sles,test_update,test_sles_smt_reg,test_sles_guestregister,test_sles_on_demand,test_sles_gce';
 
 our $img_proof_tests = {
-    'Azure-BYOS'       => $azure_byos,
-    'Azure-Basic'      => $azure_on_demand,
-    'Azure-Standard'   => $azure_on_demand,
-    'Azure-CHOST-BYOS' => $azure_byos,
-    'Azure-HPC'        => $azure_on_demand,
-    'Azure-HPC-BYOS'   => $azure_byos,
-
-    'EC2-CHOST-BYOS'   => $ec2_byos_chost,
-    'EC2-HVM'          => $ec2_on_demand,
-    'EC2-HVM-ARM'      => $ec2_on_demand,
-    'EC2-HVM-BYOS'     => $ec2_byos,
-    'EC2-HVM-HPC-BYOS' => $ec2_byos,
-
-    GCE              => $gce_on_demand,
-    'GCE-BYOS'       => $gce_byos,
-    'GCE-CHOST-BYOS' => $gce_byos,
+    'Azure-BYOS'         => $azure_byos,
+    'AZURE-BYOS-Updates' => $azure_byos,
+    'Azure-Basic'        => $azure_on_demand,
+    'Azure-Standard'     => $azure_on_demand,
+    'Azure-CHOST-BYOS'   => $azure_byos,
+    'Azure-HPC'          => $azure_on_demand,
+    'Azure-HPC-BYOS'     => $azure_byos,
+
+    'EC2-CHOST-BYOS'       => $ec2_byos_chost,
+    'EC2-HVM'              => $ec2_on_demand,
+    'EC2-HVM-ARM'          => $ec2_on_demand,
+    'EC2-HVM-BYOS'         => $ec2_byos,
+    'EC2-HVM-BYOS-Updates' => $ec2_byos,
+    'EC2-HVM-HPC-BYOS'     => $ec2_byos,
+
+    GCE                => $gce_on_demand,
+    'GCE-BYOS'         => $gce_byos,
+    'GCE-BYOS-Updates' => $gce_byos,
+    'GCE-CHOST-BYOS'   => $gce_byos,
 };
 
 sub run {
-    my ($self) = @_;
+    my ($self, $args) = @_;
 
     $self->select_serial_terminal;
 
-    my $flavor   = get_required_var('FLAVOR');
-    my $provider = $self->provider_factory();
-    my $instance = $provider->create_instance();
-    my $tests    = get_required_var('PUBLIC_CLOUD_IMG_PROOF_TESTS');
+    my $flavor = get_required_var('FLAVOR');
+    my $tests  = get_required_var('PUBLIC_CLOUD_IMG_PROOF_TESTS');
+    my $provider;
+    my $instance;
+
+    # QAM passes the instance as argument
+    if (get_var('PUBLIC_CLOUD_QAM')) {
+        $instance = $args->{my_instance};
+        $provider = $args->{my_provider};
```

Comment:
Regarding cleanup, AFAIC the simplest is here to set it like 
```
$self->{provider} = $args->{my_provider};
```
And then `publiccloud::basetest::_cleanup()` should do the job. But remember, the instance is then terminated after this module.

Reply:
I've added your suggestion to `img_proof.pm`